### ISSUE TITLE
Exception: unsupported audio type error

### ISSUE THREAD

**Issue by ShinGitS** [gchudov/cuetools.net#87]
![cuetool218-encoding](https://user-images.githubusercontent.com/78285707/109705802-06b2a000-7b4d-11eb-9b05-aad0e2940955.jpg)

doing a test encode of a flac.   2.16 has no issues reading and writing files.   It is a stock installation of 2.18HF1.  Only thing changed was the output.

**Comment by c72578** [comment]
@ShinGitS Thanks for your report.
Could you please post the output of the following command, which shows details about the .flac file:
```console
cmd
ffprobe.exe "D:\Music\OST\Video Games\Starcraft 2 Wings of Liberty OST\01. Glenn Stafford, Derek Duke, Russel Brower, Neal Acree - Wings of Liberty.flac"
```
Remark: **`ffprobe.exe`** can be obtained e.g. from here:
https://github.com/ffbinaries/ffbinaries-prebuilt/releases/download/v4.2.1/ffprobe-4.2.1-win-64.zip

**Comment by c72578** [comment]
@ShinGitS Please also have a look at **Known Issues When Unpacking the Archive**
http://cue.tools/wiki/CUETools_Download#Known_Issues_When_Unpacking_the_Archive
"Unsupported audio type" can occur when a decoder is not found or CUETools cannot access a decoder or the plugins folder.

**Comment by ShinGitS** [comment]
sorry about the delay.  And my apologies in regards to renaming the file.  I could not get it to "find" file.  The input path kept returning "no file"  so I just copied to where I stored ffprobe, and renamed the file down to wings.flac.   I assure you that it is the same file. 
![ffprobe-output](https://user-images.githubusercontent.com/78285707/109732157-78044a00-7b71-11eb-8d86-7765c0f63969.jpg)

I did run it unblocked.

**Comment by c72578** [comment]
Thanks for checking.
Does it work for you with „normal“ flac files? Meaning flac files without an additional mjpeg video stream?

**Comment by c72578** [comment]
@ShinGitS Could you please post a screenshot of **`CUETools - Settings - Decoders`**? To check if the flac decoders are available on your system.
It should look like this:
![CUETools_Settings_Decoders](https://user-images.githubusercontent.com/371551/109863856-6b135380-7c62-11eb-9d51-feabdcc8c284.PNG)

**Comment by ShinGitS** [comment]
Ok.  I think i see the problem.  Thus it raises a different problem.   I just extract the 2.18hf1 compressed file into its own directory.  It is as they call it "stock"  So I do not understand the difference in your screenshot and mine.  

![cuetools218decoder](https://user-images.githubusercontent.com/78285707/110033779-ac514500-7cee-11eb-8931-23d42ce69e16.jpg)

**Comment by gchudov** [comment]
Plugin DLLs failed to load - could be they weren't unpacked properly (e.g. when trying to run CUETools.exe from the archive without unpacking it) or could be some kind of anti-virus software preventing dynamic library loading from that location.

**Comment by gchudov** [comment]
CUETools.exe.config seems to be missing from archive

**Comment by ShinGitS** [comment]
Ok I used RAR to extract the files.  So the file structure looks like this. F:\tmp\cuetools_2.1.8_hf1\cuetoold_2.1.8.   So it has been extracted exactly as the structure in the zipfile.  

I have unblocked it.  Once again it is  similar to the screen shot I have posted above.   I have given it admin access, it still shows what the above screen shot has been given.    

The drive is not a network drive.  It is a physical drive on my system.  Even with my AV software (only MS Defender running),  it shows the same thing.

**Comment by c72578** [comment]
@ShinGitS Two further things to investigate please:
1.) Could you please run the following command using **`cmd`** and upload the generated **`dir_cuetools.txt`** file here?
```console
dir /s F:\tmp\CUETools_2.1.8_HF1\CUETools_2.1.8 >F:\tmp\dir_cuetools.txt
```
2.) Please check, if there are any blocked dll files left in the plugins dir, e.g.:
![dll_blocked_red](https://user-images.githubusercontent.com/371551/110069506-3db8c980-7d78-11eb-98e8-88290c6c7607.png)


Just in case, running this command from **`powershell`** will unblock any of them recursively:
```console
dir -Path F:\tmp\CUETools_2.1.8_HF1\CUETools_2.1.8 -Recurse | Unblock-File
```

**Comment by ShinGitS** [comment]
Sorry about the tardiness of the reply and subsequent turnaround time.   As far as I can tell,  the file/directory structure seems to be right,  everything is unblocked. 

[dir_cuetools.txt](https://github.com/gchudov/cuetools.net/files/6095928/dir_cuetools.txt)

**Comment by c72578** [comment]
OK, thanks for checking the file/directory structure. It is, as it should be.
So far, I could only reproduce the issue described here, if I delete the **`plugins`** subdirectory completely ...

- Could you please provide the following additional details here:
  - Windows version, including buildnumber and bitness, e.g. using **`cmd`**:
  ```
  wmic os get Caption, Version, BuildNumber, OSArchitecture
  ```
  - A list of installed .NET versions. This can be obtained e.g. using **`powershell`**:
  ```powershell
  Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP' -Recurse | Get-ItemProperty -Name version -EA 0 | Where { $_.PSChildName -Match '^(?!S)\p{L}'} | Select PSChildName, version
  ```
- Have you already tried adding a folder exclusion to your Antivirus software for `F:\tmp\CUETools_2.1.8_HF1\CUETools_2.1.8`?

**Comment by ShinGitS** [comment]
Sorry, weekend :D  .  Here is version of both .net and my OS.  Yes I have put in the exclusion into my AV software.   I am fairly certain that for some reason it just doesn't see/link/call the files properly.  I can always fall back to 2.16 for now.  It is just puzzling that is all.  

![os and net version](https://user-images.githubusercontent.com/78285707/110378611-a6b77000-800a-11eb-96e3-24ec9f177e36.jpg)

**Comment by c72578** [comment]
@ShinGitS thanks for providing the details concerning the OS and .NET versions. Looks all good.
- What about CUETools 2.1.7 - is it working properly on your system - like 2.1.6?
- Did you already try CUETools 2.1.8, when it has been unzipped to a folder on the C:\ drive?
- In the CUETools folder, there is a file called "user_profiles_enabled". Can you try CUETools 2.1.8 after renaming this file to "user_profiles_disabled". This will create fresh settings inside the CUETools_2.1.8 folder and will not use the settings from the user profile folder.

**Comment by ShinGitS** [comment]
Sorry for the delay.  

- Cuetools 2.17: I have not tried it.  Used 2.16, and completely missed the 2.17 release.  I will download and try it sometime this week. 
- I keep my C drive mainly meant as an OS drive + select applications.  Most of my installation goes into either my D drive or F drive.  I can put it into one of the drives root directory.  Just wondering why and if it would have any effect.
- I will rename the file after trying 2.17.   

Thanks in advance for spending so much of your valuable time on this issue.

**Comment by darkred** [comment]
I'm having the exact same problem.
I'd like to provide all the asked information, hoping it will assist you in fixing this *(click each one to expand)* :

<details>
<summary>Windows version, including buildnumber and bitness, e.g. using <code>cmd
</code>:</summary>

```
wmic os get Caption, Version, BuildNumber, OSArchitecture

BuildNumber  Caption                   OSArchitecture  Version
19042        Microsoft Windows 10 Pro  64-bit          10.0.19042
```
</details>


<details>
<summary>A list of installed .NET versions. This can be obtained e.g. using <code>powershell
</code>:</summary>

```
Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP' -Recurse | Get-ItemProperty -Name version -EA 0 | Where { $_.PSChildName -Match '^(?!S)\p{L}'} | Select PSChildName, version

PSChildName                      Version
-----------                      -------
v2.0.50727                       2.0.50727.4927
v3.0                             3.0.30729.4926
Windows Communication Foundation 3.0.4506.4926
Windows Presentation Foundation  3.0.6920.4902
v3.5                             3.5.30729.4926
Client                           4.8.04084
Full                             4.8.04084
Client                           4.0.0.0
```
</details>

<details>
<summary>ffprobe <file></summary>

```
ffprobe version 4.2.1 Copyright (c) 2007-2019 the FFmpeg developers
  built with gcc 9.1.1 (GCC) 20190807
  configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Input #0, flac, from 'E:\MP3 VIDEOS  NEW\Stephan Forte - The Shadows Compendium (KICP 1600)\Stephan Forte - The Shadows Compendium.flac':
  Metadata:
    Genre           : Metal
    Artist          : Stephan Forte
    Title           : De Praestigiis Daemonum
    Album           : The Shadows Compendium
    Date            : 2011
    track           : 02
    Comment         : Exact Audio Copy
  Duration: 00:48:50.03, start: 0.000000, bitrate: 912 kb/s
    Stream #0:0: Audio: flac, 44100 Hz, stereo, s16
```
</details>

The `CUETools_2.1.6/7/8` directories are all located in `C:\`:
![2021-03-15_125233](https://user-images.githubusercontent.com/723651/111142427-581b5000-858d-11eb-8cba-b874a6c3958b.jpg)

CUETools 2.1.6 and 2.1.7 both work properly in on my system

Renaming in `C:\CUETools_2.1.8` the "user_profiles_enabled" to "user_profiles_disabled" doesn't help.

I was getting:
![2021-03-15_125547](https://user-images.githubusercontent.com/723651/111142957-f0b1d000-858d-11eb-989d-37c27de1f9fc.jpg)
because there blocked dll files left in the plugins dir.
So, I extracted CUETools_2.1.8_HF1.zip via winrar, 
*(alternatively I'd use Sysinternals `streams -s -d` to unblock all contained files manually)*,
and now the files are unblocked, but now I'm getting:
`Exception: flake.exe "The system cannot find the file specified"`

![2021-03-15_132013](https://user-images.githubusercontent.com/723651/111145947-8bf87480-8591-11eb-81d2-af16952119c7.jpg)

And three 2.1.8 screenshots:

<details>
<summary>Details</summary>

[CUETools main window](https://user-images.githubusercontent.com/723651/111146576-47210d80-8592-11eb-9c6d-ea17b520716a.jpg)
[Encoders|flake.exe [flac]](https://user-images.githubusercontent.com/723651/111146675-60c25500-8592-11eb-94ca-c6fc91493a41.jpg)
[Decoders|cuetools [flac]](https://user-images.githubusercontent.com/723651/111146701-66b83600-8592-11eb-90d6-d57a41ccc0ec.jpg)


</details>

**Comment by c72578** [comment]
@darkred thanks for your report.
You can download flake from here (Flake 0.11 Windows binary):
http://flake-enc.sourceforge.net/download.html
Just unpack **`flake.exe`** to `C:\CUETools_2.1.8` and it will be found.

**Comment by darkred** [comment]
Adding that file fixes the issue. Thank you very much!

But, I did a different thing:
deleted the `C:\CUETools_2.1.8` folder itself , 
as well as the old `CUE Tools` and `CUERipper` folders found in the %appdata% folder.

Then extracted `CUETools_2.1.8_HF1.zip` again, and deleted the extracted `user_profiles_enabled` file,
so that settings will be saved to the same folder instead of my user profile folder.

No flake.exe needed, it uses the `flake [cuetools]` instead of looking for the missing `flake.exe [flac]`

**Comment by c72578** [comment]
@ShinGitS are you still experiencing this issue?
In the meantime, all files of **`CUETools 2.1.8`** are accepted automatically by **`Windows Defender SmartScreen`** using current definitions and no file is blocked anymore.
For this purpose, the zip files have been submitted to Microsoft and analyzed here:
https://www.microsoft.com/en-us/wdsi/filesubmission

Submitted and analyzed files:
**`CUETools_2.1.8_HF1.zip, CUETools_2.1.8.zip, CUETools.CTDB.EACPlugin.2.1.8.zip`**

**Comment by ShinGitS** [comment]
yes sorry, I have been meaning to get around to testing.  Unfortunately work has gotten in the way.   I have 2.17 downloaded, just need time to test.  this weekend.

**Comment by ShinGitS** [comment]
Sorry for the late response.  
2.17 does not have the same issues.   I redownloaded the 2.18 that has been accepted by smart screen.  Now it is working as intended.   There is a separate issue of invalid:stripe  but that is something for another day.  

Thank you once again for taking all the time to bug fix the issue.   It was probably some file is locked/blocked and not allowed to be accessed.

**Comment by c72578** [comment]
Thanks for checking and providing the info.

**Comment by erjok** [comment]
I have the same exception "Unsupported audio type: XXX.flac"
I believe it's related to missing decoders. Tried different app versions 2.17, 2.18, 2.2 but the app always have only three decoders:

- cuetools [wav]
- takc.exe [take]
- ffmpeg.exe [m4a]

I see a lot of files in the plugins folder (e.g. CUETools.Codecs.libFLAC.dll). Non of these files are blocked but looks like the app doesn't use them anyway.

**Comment by rramstad833** [comment]
I'll echo the previous poster.  I am on Windows 10 x64 and have been working with 2.2.1 and getting nowhere.

Initially I just unpacked it and tried to run it, only three decoders.

After reading this thread, I deleted the appropriate stuff in %appdata% and the folder and re-extracted it and changed to user_profiles_disabled before running it for the first time, and same result, only three decoders.

I am not seeing any of that stuff regarding Unblocking anywhere, no check box, no language like that at all.

**Comment by c72578** [comment]
@rramstad833 Thanks for your report.
Could you please add the following information:
- Which antivirus software are you using?
- Did you unblock the downloaded CUETools_2.2.1.zip before unpacking it?
- Which software did you use to unpack CUETools_2.2.1.zip?

**Comment by rramstad833** [comment]
Sorry, apparently apologies are in order.

If the file has been recently unpacked, and you right click on the ZIP file
in Windows 10, the Unblock checkbox doesn't appear at all.  This is strange
but true.

Beyond that, when it does appear, the wording is confusing.  Checking the
box unblocks it -- if the box is left clear, which logically should be
unblocking it, it is actually blocked.

For reference, I am not using any antivirus software.  There is an included
antivirus system within Windows 10, which is enabled.  I used the internal
Windows 10 tool that is part of Windows Explorer to unpack the ZIP file.

In short, if you do what I did -- unpack the ZIP file, find that I can't
get FLAC files to work, search the internet for solutions, get told to
check the Unblock box -- there is no Unblock box, as it disappears if
you've extracted the ZIP file recently.  Super weird.

I was able to get FLAC and other decoders to appear by deleting the
extracted files, deleting %appdata% for the tool, then doing the right
click Unblock on the ZIP, opening up the extracted folder, changing
user_profiles_enabled to user_profiles_disabled

Again, my apologies.  Seems like the instructions to fix the problem might
need to include the additional steps of cleaning up any previous
extraction, as that seemed to be the key to get the Unblock checkbox to
appear.

**Comment by c72578** [comment]
@rramstad833 Thanks for the details.
Could you please try the following:
- Delete the unpacked CUETools_2.2.1 directory
- Unpack CUETools_2.2.1.zip using a different program, like [7-Zip](https://www.7-zip.org/), Total Commander etc.

Using different unpacking programs works around the blocking of unpacked files.

**Comment by rramstad833** [comment]
I did get it to work using the Windows 10 built in unzip by deleting the
folders in %appdata% and the extracted directory, then extracting again
with Unblock checked, and then editing the filename changing
user_profiles_enabled to user_profiles_disabled before running it for the
first time.

I don't have other unpacking programs and can't help with further
debugging.  I suspect you are correct that a different program might work
right off the bat.

The super confusing part for me is that the Unblock option didn't appear
once I had unpacked the directory and had trouble.  So, I go online to look
for solutions and am told to check a box that doesn't appear!

The solutions online should probably tell people to A) delete the extracted
directory B) delete any user settings in %appdata% and then C) right click
the ZIP and check the box to Unblock.

If you are having problems and simply right click the ZIP the option to
Unblock doesn't appear.  (I assume this is some security thing from
Microsoft...  or just flaky Windows 10 behavior.)

Hope this helps.

**Comment by c72578** [comment]
@rramstad833 Good to hear, that it is working for you now.

Just for the records, here is an alternative unpacking approach using a Windows 10 builtin utility from the command line:
`tar -xf CUETools_2.2.1.zip`

This will unpack CUETools_2.2.1.zip and avoid blocking of extracted files.
Remark: `tar.exe` is available in Windows 10 since Version 1803 (April 2018 Update)

**Comment by tzr916** [comment]
Took me several hours to get out of this rabbit hole! I have been running 2.1.6 in Windows 10 for the longest time, decided to give try 2.2.2 but I could not get past this unsupported error. Tips to getting it working:
1. I looked at the "blocked" checkbox but it was NEVER checked, so this was not the issue.
2. Do NOT use the windows 10 default unzip program, use 7-Zip.

After unzipping but BEFORE running cuetools.exe for the first time:
1. Delete your %appdata% cuetools folder, and rename user_profiles_enable to user_profiles_disable.
2. If you are relocating the installation folder to the C:\Program Files (x86) directory, you will need to run cuetools.exe as admin and/or create a shortcut that runs it as admin.

**Comment by freonpsandoz** [comment]
Run as administrator?? 2.1.6 didn't require any of this nonsense. Didn't the developer test 2.2? Installing a Windows program isn't supposed to require the hundreds of hours of user experimentation described here.

**Comment by AntonioRedondo** [comment]
I came across the same issue with version 2.2.6 on Windows 11 Pro 24H2. Bug tedious enough for many people to give up before finding a workaround. But @tzr916 fix worked for me.
